index.bs

```diff
@@ -825,23 +831,45 @@ When this method is invoked, the user agent MUST execute the following algorithm
         <code>|options|.{{PublicKeyCredentialRequestOptions/allowCredentials}}.{{PublicKeyCredentialDescriptor/type}}</code>.
         Set |allowCredentialDescriptorList| to this filtered list.
 
-    1. If |allowCredentialDescriptorList| 
+    1. If |allowCredentialDescriptorList|
         <dl class="switch">
             :   [=list/is not empty=]
-            ::  [=in parallel=], [=list/for each=] credential descriptor |C| in |allowCredentialDescriptorList|:
-                1. If <code>|C|.{{transports}}</code> [=list/is not empty=], the client SHOULD select one |transport| from
-                    {{transports}}. Then, using |transport|, invoke the [=authenticatorGetAssertion=] operation on
-                    |authenticator|, with |rpId|, |clientDataHash|, |allowCredentialDescriptorList|, and
-                    |authenticatorExtensions| as parameters.
-    
-                1. Otherwise, using local configuration knowledge of the appropriate transport to use with |authenticator|, 
-                    invoke the [=authenticatorGetAssertion=] operation on |authenticator| with |rpId|,
-                    |clientDataHash|, |allowCredentialDescriptorList|, and |clientExtensions| as parameters.
+            ::  1. Let |distinctTransports| be a new [=ordered set=].
+
+                1. If |allowCredentialDescriptorList| has exactly one value, let |savedCredentialId| be a new {{ArrayBuffer}},
+                       created using |global|'s [=%ArrayBuffer%=], and containing the bytes of
+                       <code>|allowCredentialDescriptorList|[0].id</code>.
+
+                1. [=list/For each=] credential descriptor |C| in |allowCredentialDescriptorList|,
+                    [=set/append=] each value, if any, of <code>|C|.{{transports}}</code> to |distinctTransports|.
+
+                    Note: This will aggregate only distinct values of {{transports}} (for this [=authenticator=]) in
+                        |distinctTransports| due to the properties of [=ordered sets=].
+
+                1. If |distinctTransports|
+                    <dl class="switch">
+                        :   [=list/is not empty=]
+                        ::  The client selects one |transport| value from |distinctTransports|, possibly incorporating local
+                            configuration knowledge of the appropriate transport to use with |authenticator| in making its
+                            selection.
+
+                            Then, using |transport|, invoke [=in parallel=] the [=authenticatorGetAssertion=] operation on
```

Comment:
actually, as suggested in https://github.com/w3c/webauthn/issues/472#issuecomment-309871552 (mentioned above), we will remove those [=in parallel=] invocations because we are already on a background thread given how we are invoked from https://w3c.github.io/webappsec-credential-management/#algorithm-request